src/hooks/useUpdateSubscription.tsx

```diff
@@ -0,0 +1,40 @@
+import { useCallback, useContext } from 'react';
+import { TokenContext } from 'hooks/useCustomerToken';
+import { useAsyncOnDemand } from 'hooks/useAsync';
+import type { UseAsyncOnDemandReturnType } from 'hooks/useAsync';
+import subscribeCustomer from 'actions/subscribeCustomer';
+import type { SubscribeCustomerOptions } from 'actions/subscribeCustomer';
+import { components } from '../apiTypes';
+
+export type ActiveSubscription =
+  components['schemas']['CustomerPortalSubscription'];
+
+type Props = {
+  token?: string;
+  pricePlanName: string;
+  options?: SubscribeCustomerOptions;
+};
+
+const useUpdateSubscription = (
+  props: Props
+): UseAsyncOnDemandReturnType<ActiveSubscription> => {
+  const { token: tokenFromContext } = useContext(TokenContext);
+  const { token, pricePlanName, options = {} } = props;
```

Comment:
Since you're not doing anything with `props` besides destructuring it, you could do the destructuring in the function parameters instead:

```ts
const useUpdateSubscription = (
  { token, pricePlanName, options = {} } : Props
  ) : UseAsyncOnDemandReturnType<ActiveSubscription> => { ...
```